### ISSUE TITLE
Unknown column 'availability' in 'field list' Moodle4.0.4 mod_booking7.5.0

### ISSUE THREAD

**Issue by baltaner** [Wunderbyte-GmbH/moodle-mod_booking#212]
Hi,
after upgrading mod_booking from 7.2.7 to 7.5.0 I get a database error.

ubuntu 22.04.1, mariadb 10.6.7, php7.4

I fixed it - probably not correct - with:
ALTER TABLE `mdl_booking_options` ADD `availability` TEXT NOT NULL AFTER `dayofweek`;

Below the debug log

Thanks 
Bertold

```
Edit booking option
Error reading from database

More information about this error

Debug info: Unknown column 'availability' in 'field list'
SELECT availability FROM mdl_booking_options WHERE id = ?
[array (
0 => 543,
)]
Error code: dmlreadexception×Dismiss this notification
Stack trace:
line 486 of /lib/dml/moodle_database.php: dml_read_exception thrown
line 291 of /lib/dml/moodle_read_slave_trait.php: call to moodle_database->query_end()
line 1273 of /lib/dml/mysqli_native_moodle_database.php: call to mysqli_native_moodle_database->query_end()
line 1671 of /lib/dml/moodle_database.php: call to mysqli_native_moodle_database->get_records_sql()
line 1744 of /lib/dml/moodle_database.php: call to moodle_database->get_record_sql()
line 1725 of /lib/dml/moodle_database.php: call to moodle_database->get_field_sql()
line 1704 of /lib/dml/moodle_database.php: call to moodle_database->get_field_select()
line 728 of /mod/booking/classes/form/option_form.php: call to moodle_database->get_field()
line 357 of /mod/booking/editoptions.php: call to mod_booking\form\option_form->set_data()

```

**Comment by dasistwas** [comment]
The plugin does not support Moodle 4 yet. But thank you for reporting this bug. We will investigate that.

**Comment by baltaner** [comment]
Hi,
I know that - I use mod_booking since version 7.0.26 with moodle4.x - it works for us very well!
Thank you very much for your great work!

**Comment by bernhard-wunderbyte** [comment]
@baltaner Thank you so much for reporting this!

There was a problem with the upgrade script. I already fixed it. The new version containing the fix is: v7.5.2 (Please remove the "availability" column from your <prefix>_booking_options table before installing the new version, so it will get updated properly.
